src/net.ts

```diff
@@ -1,23 +1,37 @@
+import * as path from 'path';
+import * as fs from 'fs';
+import semver from 'semver';
 import Provider from './provider';
+const VERSIONS_PATH = path.resolve(__dirname, './protocol_versions.json');
+const SDK_VERSION = require('./package.alias.json').version;
 
 export default class Network {
   public provider: Provider;
+  public protoVer: string;
+  // public version: string;
 
   /**
    * @param {Provider} provider
    * @constructor
    */
   constructor (provider: Provider) {
     this.provider = provider;
-    // TODO (lia): should get the following info from the node
+    if (!fs.existsSync(VERSIONS_PATH)) {
+      throw Error('Missing protocol versions file: ' + VERSIONS_PATH);
+    }
+    const VERSION_LIST = JSON.parse(fs.readFileSync(VERSIONS_PATH, 'utf-8'));
+    if (!VERSION_LIST[SDK_VERSION]) {
+      throw Error("Current sdk version doesn't exist in the list");
+    }
+    this.protoVer = VERSION_LIST[SDK_VERSION].max;
```

Comment:
Would it be helpful to leave some comments on the strategy (i.e., use the max value first and try to adapt depending on the server's protoVer if it's too high)?

Reply:
Good point–left some comments.